sdv/sampling/hierarchical_sampler.py

```diff
@@ -68,7 +68,7 @@ def _sample_rows(self, synthesizer, num_rows=None):
                 Sampled rows, shape (, num_rows)
         """
         num_rows = num_rows or synthesizer._num_rows
-        return synthesizer._sample_batch(num_rows, remove_extra_columns=False)
+        return synthesizer._sample_batch(int(num_rows), keep_extra_columns=True)
```

Comment:
@amontanez24 @pvk-developer This line is an example where using an assert could be useful. `num_rows` should always be an integer, and if it is not then wherever it was set did so incorrectly, so the code there should be updated to cast it to an int.

So at this point we should `assert isinstance(num_rows, int)`, or change the docstring to say that `num_rows` can be a float as well. You shouldn't have to write code assuming other methods/inputs are invalid, ie casting to `int` in this case.